src/util/cached_chain.rs

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Utilities for adhering to a cached SEV chain convention.
+//!
+//! The search path for the SEV chain is:
+//!   1. The path specified in the "SEV_CHAIN" environment variable
+//!      (if present).
+//!   2. `$HOME/.cache/amd-sev/chain`
+//!   3. `/etc/cache/amd-sev/chain`
+//!
+//! An entire certificate chain can be created using the `sevctl`
+//! utility.
+
+use std::env;
+use std::fs::File;
+use std::io::{ErrorKind, Result};
+use std::path::{Path, PathBuf};
+
+use codicon::Decoder;
+
+use crate::certs::Chain;
+
+fn append_rest<P: AsRef<Path>>(path: P) -> PathBuf {
+    let mut path = path.as_ref().to_path_buf();
+    path.push("amd-sev");
+    path.push("chain");
+    path
+}
+
+/// Returns the path stored in the optional `SEV_CHAIN`
+/// environment variable.
+pub fn env_var() -> Option<PathBuf> {
+    env::var("SEV_CHAIN").ok().map(PathBuf::from)
+}
+
+/// Returns the "user-level" search path for the SEV
+/// certificate chain (`$HOME/.cache/amd-sev/chain`).
+pub fn home() -> Option<PathBuf> {
+    dirs::cache_dir().map(append_rest)
+}
+
+/// Returns the "system-level" search path for the SEV
+/// certificate chain (`/etc/cache/amd-sev/chain`).
```

Comment:
```console
# dnf install /etc/cache
No match for argument: /etc/cache
```

You are introducing a new subdir in `/etc`. I think you want `/var/cache`, which already exists.

Reply:
Or use `/etc/amd-sev` directly.

Reply:
Good catch, thanks. Switched to using `/var/cache`